### PR TITLE
My account features

### PR DESCRIPTION
**Pull request by maximebiloe**
Add some styles and the following features in my-account :
- Order history
- Order details
- Order return
- Credit slips
- Guest tracking

### CODE REVIEW

File: classes/pdf/PDF.php [hunk 1]
```diff
@@ -51,13 +51,14 @@ public function __construct($objects, $template, $smarty, $orientation = 'P')
     {
         $this->pdf_renderer = new PDFGenerator((bool)Configuration::get('PS_PDF_USE_CACHE'), $orientation);
         $this->template = $template;
+        $smarty->escape_html = false;
```
Comment (djfm) [inline classes/pdf/PDF.php]: Is `$smarty` the global smarty object here? If so, I'm afraid you're actually disabling the autoescape for everything and not just the PDF!

File: controllers/front/GuestTrackingController.php [hunk 2]
```diff
@@ -111,94 +94,61 @@ public function postProcess()
...
         if (count($this->errors)) {
             sleep(1);
         }
 
-        $this->context->smarty->assign(array(
-            'action' => $this->context->link->getPageLink('guest-tracking.php', true),
-            'errors' => $this->errors,
-        ));
-        $this->setTemplate(_PS_THEME_DIR_.'guest-tracking.tpl');
+        // $this->context->smarty->assign(array(
```
Comment (djfm) [inline controllers/front/GuestTrackingController.php]: Leaving a few traps for future developers? :) If commented code is useless go ahead and remove it!

File: controllers/front/OrderDetailController.php [hunk 3]
```diff
@@ -108,36 +106,38 @@ public function postProcess()
...
+                        );
                     }
 
-                    if (Tools::getValue('ajax') != 'true') {
-                        Tools::redirect('index.php?controller=order-detail&id_order='.(int)$idOrder);
-                    }
+                    $this->success[] = $this->l('Message successfully sent');
 
-                    $this->context->smarty->assign('message_confirmation', true);
+                    // if (Tools::getValue('ajax') != 'true') {
```
Comment (djfm) [inline controllers/front/OrderDetailController.php]: Is this commented code necessary?

**Comment by djfm** [comment]
I think there is something weird going on with price conversions: I have 2 currencies installed and when I change the currency then in my order history only the currency unit changes - not the amounts.

In EUR:

image

In USD:

image

I've set up my shop to have 1 EUR = 2 USD so there should be a difference if the amount were converted.

Current behaviour is to display everything in the currency the order was placed with, and I think it makes sense. If I had to guess I'd say the issue is related to the use of Adapter_PricePresenter, which takes the currency from the global context. I think you should use the order's currency everywhere.

**Comment by julienbourdeau** [comment]
Here are my thoughts:

- [ ] Nest all order data inside the `$order` array. Like in the template instead of using `$products` you'll be using `$order.products`. For customer for example, you'll avoid conflicts with the customer defined by `FrontController::getTemplateVarCustomer()` 
- [ ] Add parameter to `FrontController::getTemplateVarCustomer()` so you can use this method [here](https://github.com/PrestaShop/PrestaShop/pull/4308/files#diff-ab7196159bfc10c261265a92ecc25b66R139) (all customer passed to the template will be consistent).
- [ ] I think some errors should be redirects or 404, like [this error](https://github.com/PrestaShop/PrestaShop/pull/4308/files#diff-6526e9279c69dc91f6035edad590a212R150) or [this error](https://github.com/PrestaShop/PrestaShop/pull/4308/files#diff-6526e9279c69dc91f6035edad590a212R177).
- [ ] I would remove most `title="{l s='xxx'}"`, it will make theme translation longer and harder and it doesn't help much ([example](https://github.com/PrestaShop/PrestaShop/pull/4308/files#diff-2f43d4fb175d9d47ca8f184c3999ff29R35)).
- [ ] I just realized we should use a partial [for these buttons](https://github.com/PrestaShop/PrestaShop/pull/4308/files#diff-231be4031edeceb0290d44ba4f432af6R34) across the customer area

And agree we should only use the currency of the order, not the one from the context.

**Comment by Maciej-Kaminski** [comment]
Dear Julien, 
From the customer point of view,  order history is useless, because people need to open every order too search products. This is very frustrating with virtual ones. 
Maybe you could think to add filter,  using product ID as key, not order name.

**Pull request merged**